#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/Floor.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/VarnishedPine.proto"

WorldInfo {
  basicTimeStep 8
}
Viewpoint {
  orientation -0.4733058455123405 -0.46854800500668004 0.7459519713816088 4.413739092437097
  position 1.0329545986170547 2.167131518897487 2.9980688817797194
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF robot_1 Robot {
  translation 1.01615 1.5879 0.0399299
  rotation -7.071209999558151e-06 8.65847999945897e-06 -0.9999999999375144 1.55232
  children [
    Solid {
      translation 0.095 0 0.04
      children [
        DEF cuerpo_camara Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.03
          }
        }
      ]
      name "cuerpo_camara"
      boundingObject USE cuerpo_camara
      physics Physics {
      }
    }
    Camera {
      translation 0.095 0 0.06
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      width 1200
      height 1200
      antiAliasing TRUE
      recognition Recognition {
        frameThickness 10
      }
    }
    DistanceSensor {
      translation 0.08 0.04 -0.027
      rotation 0 1 0 1.5707996938995747
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            metalness 0
          }
          geometry Box {
            size 0.005 0.01 0.01
          }
        }
      ]
      name "ir_left"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.08 -0.04 -0.027
      rotation 0 1 0 1.5707996938995747
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            metalness 0
          }
          geometry Box {
            size 0.005 0.01 0.01
          }
        }
      ]
      name "ir_right"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.1 0.04 0
      rotation 0 0 1 0.26180030610042526
      children [
        DEF sensor_shape Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      boundingObject USE sensor_shape
      physics Physics {
      }
      recognitionColors [
        0 0 0
      ]
      lookupTable [
        0 0 0
        0.3 30 0
      ]
    }
    DistanceSensor {
      translation 0.1 0 0
      children [
        DEF sensor_shape Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_center"
      boundingObject USE sensor_shape
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.29 30 0
      ]
    }
    DistanceSensor {
      translation 0.1 -0.04 0
      rotation 0 0 1 -0.2617996938995747
      children [
        DEF sensor_shape Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
      boundingObject USE sensor_shape
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.3 30 0
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 15.78077345392575
        axis 0 1 0
        anchor 0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint DEF Wheel1 Solid {
        translation 0.04999969605341672 0.06 -4.3181551176192185e-08
        rotation -0.025720047119908294 0.7068714535664469 -0.7068742655585987 3.1930212063418946
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        name "solid wheel1"
        boundingObject USE Wheel
        physics DEF Wheel_PH Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 7.49441556919439
        axis 0 1 0
        anchor 0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint DEF Wheel2 Solid {
        translation 0.049998291147553 -0.059999999566905826 -9.641930468667455e-07
        rotation 0.714519105066131 0.4947023349726672 -0.4947040006590668 1.9007878161276481
        children [
          USE Wheel
        ]
        name "wheel2"
        boundingObject USE Wheel
        physics USE Wheel_PH
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 15.780784174292197
        axis 0 1 0
        anchor -0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint DEF Wheel3 Solid {
        translation -0.05000016040991377 0.05999999956724689 1.955329208070438e-06
        rotation 0.025723833543879464 -0.7068716005992444 0.7068739807448489 3.090156502951888
        children [
          USE Wheel
        ]
        name "wheel3"
        boundingObject USE Wheel
        physics USE Wheel_PH
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 7.494426144806155
        axis 0 1 0
        anchor -0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint DEF Wheel4 Solid {
        translation -0.05000170885577377 -0.06000000043275322 -9.641871579045822e-07
        rotation 0.7145151524260716 0.4947051894202946 -0.49470685513158025 1.9007930495878915
        children [
          USE Wheel
        ]
        name "wheel4"
        boundingObject USE Wheel
        physics USE Wheel_PH
      }
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
        IBLStrength 0
        normalMapFactor 0
        occlusionMapStrength 0
      }
      geometry Box {
        size 0.2 0.1 0.05
      }
    }
  ]
  name "robot_1"
  boundingObject USE BODY
  physics Physics {
  }
  controller "searcher"
}
Robot {
  name "supervisor"
  controller "supervisor_lines"
  supervisor TRUE
}
DEF box_1 Solid {
  translation 1.13141 1.01546 0.0499019
  rotation 6.91677891474845e-13 -8.497421404283866e-13 -1 6.045625161022676e-06
  children [
    DEF BOX_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  name "box_1"
  model "box_1"
  boundingObject USE BOX_SHAPE
  physics Physics {
    density -1
    mass 1
  }
  recognitionColors [
    0 0 1
  ]
}
DEF box_2 Solid {
  translation -1.32 0.528985 0.0499019
  rotation -1.6001639801367573e-12 8.250330037685878e-13 -1 6.045625161058373e-06
  children [
    DEF BOX_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  name "box_2"
  model "box_2"
  boundingObject USE BOX_SHAPE
  physics Physics {
    density -1
    mass 1
  }
  recognitionColors [
    1 0 0
  ]
}
DEF box_4 Solid {
  translation 0.986376 -0.0821913 0.0499019
  rotation 6.7998620164984e-13 -4.547836577542002e-13 -1 6.045649790561077e-06
  children [
    DEF BOX_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  name "box_4"
  model "box_4"
  boundingObject USE BOX_SHAPE
  physics Physics {
    density -1
    mass 1
  }
  recognitionColors [
    1 0 0
  ]
}
Floor {
  size 4 4
  tileSize 4 4
  appearance PBRAppearance {
    baseColorMap ImageTexture {
      url [
        "../mapa TFG.jpg"
      ]
    }
    metalness 0
  }
}
DEF box_3 Solid {
  translation 0.952832 0.519972 0.0499019
  rotation 1.4850011382280642e-12 3.160786422414425e-14 -0.9999999999999999 6.045625161055814e-06
  children [
    DEF BOX_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  name "box_3"
  model "box_3"
  boundingObject USE BOX_SHAPE
  physics Physics {
    density -1
    mass 1
  }
  recognitionColors [
    0 0 1
  ]
}
Solid {
  translation 0 -2 0.05
  children [
    DEF shape_muro Shape {
      appearance VarnishedPine {
      }
      geometry Box {
        size 4 0.01 0.1
      }
    }
  ]
  name "muro_sur"
  boundingObject USE shape_muro
}
Solid {
  translation -2 0 0.05
  rotation 0 0 1 1.5707903061004251
  children [
    DEF shape_muro Shape {
      appearance VarnishedPine {
      }
      geometry Box {
        size 4 0.01 0.1
      }
    }
  ]
  name "muro_oeste"
  boundingObject USE shape_muro
}
Solid {
  translation 2 0 0.05
  rotation 0 0 1 1.5707903061004251
  children [
    DEF shape_muro Shape {
      appearance VarnishedPine {
      }
      geometry Box {
        size 4 0.01 0.1
      }
    }
  ]
  name "muro_este"
  boundingObject USE shape_muro
}
Solid {
  translation -0.0698433 -0.736622 0.0499019
  rotation -0.010508592570859206 -0.021017185141725736 -0.9997238855858641 1.014231065153233e-16
  children [
    DEF BOX_SHAPE Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/factory/containers/protos/textures/wooden_box_base_color.jpg"
          ]
        }
        roughnessMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/factory/containers/protos/textures/wooden_box_roughness.jpg"
          ]
        }
        metalness 0
        normalMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/factory/containers/protos/textures/wooden_box_normal.jpg"
          ]
        }
        occlusionMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/factory/containers/protos/textures/wooden_box_occlusion.jpg"
          ]
        }
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  name "wooden box"
  model "wooden box"
  boundingObject USE BOX_SHAPE
  physics Physics {
  }
}
Solid {
  translation -0.207836 -0.412057 0.0499019
  rotation -0.010247991904096386 0.02561997976022244 -0.9996192251547682 1.0400223903518568e-16
  children [
    DEF BOX_SHAPE Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/factory/containers/protos/textures/wooden_box_base_color.jpg"
          ]
        }
        roughnessMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/factory/containers/protos/textures/wooden_box_roughness.jpg"
          ]
        }
        metalness 0
        normalMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/factory/containers/protos/textures/wooden_box_normal.jpg"
          ]
        }
        occlusionMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/factory/containers/protos/textures/wooden_box_occlusion.jpg"
          ]
        }
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  name "wooden box(1)"
  model "wooden box"
  boundingObject USE BOX_SHAPE
  physics Physics {
  }
}
DEF zona_A Solid {
  translation 0 0 0.2
  children [
    Solid {
      translation 0 0 0.05
      children [
        DEF señal Shape {
          appearance PBRAppearance {
            baseColor 1 1 0
            metalness 0
          }
          geometry Box {
            size 0.1 0.1 0.1
          }
        }
      ]
      name "señal"
      boundingObject USE señal
      recognitionColors [
        1 1 0
      ]
    }
    Solid {
      translation 0 0 -0.1
      children [
        DEF palo Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            metalness 0
          }
          geometry Cylinder {
            height 0.2
            radius 0.01
          }
        }
      ]
      name "palo"
      boundingObject USE palo
    }
  ]
  name "zona_A"
}
